libass/ass_font.c

```diff
@@ -90,77 +95,80 @@ static uint32_t convert_unicode_to_mb(FT_Encoding encoding, uint32_t codepoint)
     }
 
     // do convert
-    int output_size = WideCharToMultiByte(
-            codepage,
-            0,
-            input_buffer,
-            inbuf_size,
-            output_buffer,
-            4,
-            NULL,
-            &conversion_fail);
-    if (output_size == 0 || conversion_fail) {
+    int outbuf_size = WideCharToMultiByte(codepage, WC_NO_BEST_FIT_CHARS, input_buffer, inbuf_size,
+                                          output_buffer, sizeof(output_buffer), NULL, &conversion_fail);
+    if (outbuf_size == 0 || conversion_fail)
         return codepoint;
-    }
 
     result = 0;
-    for (size_t i = 0; i < output_size; ++i) {
+    for (size_t i = 0; i < outbuf_size; ++i) {
         result <<= 8;
         result |= (uint8_t)output_buffer[i];
     }
 
     return result;
 }
 #elif defined(CONFIG_ICONV)
+typedef struct { const char* names[8]; } EncodingList;
```

Comment:
There's no reason to expose this typedef to the whole file's scope. Please move it inside the `convert_unicode_to_mb` function.

Reply:
Done.